// ts-config/lib/commands/init-tslint.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const oclif = __importStar(require("@oclif/command"));
const path = __importStar(require("path"));
const utils_1 = require("../utils");
exports.commandName = path.basename(__filename, '.js');
// tslint:disable-next-line:no-default-export
class InitTsLint extends oclif.Command {
    async run() {
        const { flags } = this.parse(InitTsLint);
        utils_1.writeJsonFile(flags.config, {
            extends: 'ts-config',
            linterOptions: { exclude: ['**/lib/**', '**/node_modules/**'] }
        }, flags.force);
        return;
    }
}
InitTsLint.description = 'Creates a TSLint configuration file that extends this configuration preset.';
InitTsLint.examples = [
    `$ ts-config ${exports.commandName}`,
    `$ ts-config ${exports.commandName} --config='tslint.build.json' --force`
];
InitTsLint.flags = {
    help: oclif.flags.help({ char: 'h' }),
    config: oclif.flags.string({
        char: 'c',
        default: 'tslint.json'
    }),
    force: oclif.flags.boolean({
        char: 'f',
        description: 'overwrite an existing configuration file'
    })
};
exports.default = InitTsLint;
