// ts-config/lib/commands/init-typescript.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const oclif = __importStar(require("@oclif/command"));
const path = __importStar(require("path"));
const utils_1 = require("../utils");
exports.commandName = path.basename(__filename, '.js');
// tslint:disable-next-line:no-default-export
class InitTypeScript extends oclif.Command {
    async run() {
        const { flags } = this.parse(InitTypeScript);
        utils_1.writeJsonFile(flags.config, {
            include: ['src/**/*.ts', 'src/**/*.tsx', 'typings/**/*.d.ts'],
            compilerOptions: Object.assign({ target: 'ES2017', module: 'commonjs', moduleResolution: 'node', declaration: true, sourceMap: true, strict: true, noUnusedLocals: true, noUnusedParameters: true, noImplicitReturns: true, noFallthroughCasesInSwitch: true, forceConsistentCasingInFileNames: true, esModuleInterop: true, outDir: 'lib/' }, (flags.react
                ? {
                    lib: ['dom', 'es2017'],
                    jsx: 'react'
                }
                : {
                    lib: ['es2017']
                }))
        }, flags.force);
        return;
    }
}
InitTypeScript.description = 'Creates a TypeScript configuration file with strict settings.';
InitTypeScript.examples = [
    `$ ts-config ${exports.commandName}`,
    `$ ts-config ${exports.commandName} --config='tsconfig.build.json' --force --react`
];
InitTypeScript.flags = {
    help: oclif.flags.help({ char: 'h' }),
    config: oclif.flags.string({
        char: 'c',
        default: 'tsconfig.json'
    }),
    force: oclif.flags.boolean({
        char: 'f',
        description: 'overwrite an existing configuration file'
    }),
    react: oclif.flags.boolean({
        char: 'r',
        description: 'add React-specific settings'
    })
};
exports.default = InitTypeScript;
